components/layout/fragment.rs

```diff
@@ -1004,12 +1004,18 @@ impl Fragment {
     }
 
     /// Transforms this fragment into an ellipsis fragment, preserving all the other data.
-    pub fn transform_into_ellipsis(&self, layout_context: &LayoutContext) -> Fragment {
+    pub fn transform_into_ellipsis(&self, layout_context: &LayoutContext)
+    -> Fragment {
         let mut unscanned_ellipsis_fragments = LinkedList::new();
+        let text_overflow_string: Option<String> = match self.style().get_text().text_overflow.first {
+            longhands::text_overflow::Side::Clip => None,
```

Comment:
You should be able to make this path `unreachable!`, and return only the string in the two other paths, then remove the `unwrap()` after.